src/tactic/trunc_cases.lean

```diff
@@ -0,0 +1,41 @@
+import tactic.core
+import tactic.doc_commands
+
+namespace tactic
+namespace interactive
+
+open interactive
+open interactive.types
+open tactic
+
+/--
+Perform case analysis on a `trunc` expression,
```

Comment:
Could you say something about what the arguments `e` and `ids` are supposed to be?

Also, what is `trunc`-specific about this in the case where the target isn't a subsingleton? Isn't it basically just calling `induction`?

Reply:
I guess it is better to fail and say "just use induction"?

The one additional thing it is doing in the non-subsingleton case is unfreezing local instances if opening the `trunc` reveals a typeclass. This might seem a bit arbitrary, but was inspired by the use case in #2360. One might argue, however, that we should instead modify `induction` and `cases` to do this themselves.

Reply:
Okay, I've completely rewritten this. Really, when you have a `trunc` hypothesis, there are three things you can try:
1. hope the goal is a subsingleton, and use `induction t using trunc.rec_on_subsingleton`.
2. hope the goal doesn't depend on the hypothesis, and just use `trunc.lift_on` (this isn't compatible with `induction`, so we just `apply` it by hand)
3. just do induction, but notice that there's a useless `h_p : true` hypothesis, and in some situations if you `cases h_p` then the `eq.rec`s in the goal unravel.

The new version does the best it can of these three, and has better documentation explaining all this.

Reply:
The docs are very nice now btw, thanks!